pkg/metrics/history/internal_metrics.go

```diff
@@ -0,0 +1,99 @@
+/*
+ Licensed to the Apache Software Foundation (ASF) under one
+ or more contributor license agreements.  See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership.  The ASF licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package history
+
+import (
+	"sync"
+	"time"
+)
+
+// This class collects basic information about the cluster
+// for the web UI's front page.
+// For more detailed metrics collection use Prometheus.
+type InternalMetricsHistory struct {
+	records []*MetricsRecord
+	limit   int
+
+	// internal implementation of limited array
+	pointer int
+	full    bool
+
+	sync.RWMutex
+}
+
+type MetricsRecord struct {
```

Comment:
Do we need to export this?
IDE says there is no need to export.

Reply:
It is used in `webservice/handlers.go`, though the name `MetricsRecord` is not explicitly used. I fix it then.